<a href="https://colab.research.google.com/github/torreschavezcharlotte/Estadistica_Grupo1_2025_2/blob/main/Ejercicio_acero_grupo_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
import os

os.getcwdb()

b'/content'

In [6]:
os.listdir()

['.config', 'drive', 'sample_data']

In [7]:
os.chdir("/content/drive/MyDrive/Estadistica grupo 2 ")

In [8]:
os.getcwdb()

b'/content/drive/MyDrive/Estadistica grupo 2 '

In [9]:
os.listdir()

['acero.csv',
 'Ejercicio Produccion de Acero.docx',
 'DATOS_2025_SEM_2.xlsx',
 'Ejercicio acero grupo 2.ipynb']

In [18]:
import pandas as pd

df = pd.read_csv("acero.csv")
df

consumo  pr.tbc  pr.cc  pr.ca  pr.galv1  pr.galv2  pr.pint linea hora  \
0     135.31    6840    830      0       579      1401        0     A   1º   
1      84.08     443    903     58       611      1636      717     A   2º   
2     131.62    7270    572     36       982      1963      243     A   3º   
3      90.46    5031    694    122       896      1568        0     A   4º   
4     120.04    9365   1054    157       403      1480        0     A   5º   
..       ...     ...    ...    ...       ...       ...      ...   ...  ...   
112   113.75    9333    455    150       669       715        0     C   3º   
113   105.06    2271      0    391       697      1503        0     C   4º   
114   168.88    9625    112    325       509      1124        0     C   5º   
115   195.61   10979    700    333         0      1319        0     C   6º   
116   213.23   10441    467    248         0      1567        0     C   7º   

    temperatura averias  naverias sistema  ProdTotal     NOx      CO     COV  \
0          Alta      Si         1     OFF      11266  0.4900  3.5450  0.5450   
1          Alta      No         0     OFF       7251  0.0725  2.8950  0.4250   
2          Baja      No         0     OFF      11066  1.4900  5.0075  0.6900   
3          Baja      No         0      ON       8311  1.7150  2.1600  0.3600   
4          Baja      No         0     OFF      12459  0.4650  4.8450  0.6625   
..          ...     ...       ...     ...        ...     ...     ...     ...   
112        Alta      Si         3     OFF      11322  1.0250  3.5200  0.5225   
113        Alta      No         0     OFF       8939  0.9525  4.0050  0.5625   
114       Media      No         0      ON      13845  2.1950  5.2750  0.7450   
115        Baja      No         0     OFF      15906  1.3925  6.8275  0.9150   
116       Media      No         0      ON      17247  2.0300  7.0850  0.9475   

       SO2       CO2   N2O  
0    0.038  101.5000  6.35  
1    0.047   63.5650  2.23  
2    0.062   98.8175  5.99  
3    0.066   70.1825  3.66  
4    0.086   88.5300  6.06  
..     ...       ...   ...  
112  0.060   84.4400  4.71  
113  0.104   78.9275  5.39  
114  0.051  128.0750  7.44  
115  0.060  146.6625  8.54  
116  0.054  160.3850  8.60  

[117 rows x 20 columns]

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 20 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   consumo      117 non-null    object
 1   pr.tbc       117 non-null    int64 
 2   pr.cc        117 non-null    int64 
 3   pr.ca        117 non-null    int64 
 4   pr.galv1     117 non-null    int64 
 5   pr.galv2     117 non-null    int64 
 6   pr.pint      117 non-null    int64 
 7   linea        117 non-null    object
 8   hora         117 non-null    object
 9   temperatura  117 non-null    object
 10  averias      117 non-null    object
 11  naverias     117 non-null    int64 
 12  sistema      117 non-null    object
 13  ProdTotal    117 non-null    int64 
 14  NOx          117 non-null    object
 15  CO           117 non-null    object
 16  COV          117 non-null    object
 17  SO2          117 non-null    object
 18  CO2          117 non-null    object
 19  N2O          117 non-null    

In [19]:
df.describe()

consumo        pr.tbc        pr.cc       pr.ca    pr.galv1  \
count  117.000000    117.000000   117.000000  117.000000  117.000000   
mean   139.456496   7567.820513   295.529915  124.478632  402.803419   
std     55.185253   3002.703249   358.618556  161.990772  334.928923   
min     17.500000      0.000000     0.000000    0.000000    0.000000   
25%     99.090000   5950.000000     0.000000    0.000000    0.000000   
50%    140.070000   8808.000000    80.000000    0.000000  432.000000   
75%    182.480000   9734.000000   582.000000  248.000000  698.000000   
max    290.720000  10979.000000  1204.000000  677.000000  982.000000   

          pr.galv2     pr.pint    naverias     ProdTotal         NOx  \
count   117.000000  117.000000  117.000000    117.000000  117.000000   
mean   1159.717949  188.555556    0.675214  11841.649573    1.196132   
std     577.791602  289.445857    1.292078   4075.068276    0.643837   
min       0.000000    0.000000    0.000000   2187.000000    0.012500   
25%     933.000000    0.000000    0.000000   9023.000000    0.775000   
50%    1360.000000    0.000000    0.000000  11860.000000    1.120000   
75%    1567.000000  394.000000    0.000000  14908.000000    1.655000   
max    1963.000000  898.000000    4.000000  23202.000000    2.775000   

               CO         COV         SO2         CO2         N2O  
count  117.000000  117.000000  117.000000  117.000000  117.000000  
mean     4.377350    0.620983    0.066504  104.628141    6.114786  
std      2.095038    0.255447    0.022064   41.389709    2.125250  
min      0.250000    0.025000    0.001000   14.285000    0.870000  
25%      2.927500    0.452500    0.054000   73.717500    4.710000  
50%      4.107500    0.625000    0.066000  105.307500    6.170000  
75%      5.585000    0.785000    0.080000  134.877500    7.580000  
max     10.170000    1.255000    0.127000  218.307500   11.140000

In [25]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 20 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   consumo      117 non-null    float64
 1   pr.tbc       117 non-null    int64  
 2   pr.cc        117 non-null    int64  
 3   pr.ca        117 non-null    int64  
 4   pr.galv1     117 non-null    int64  
 5   pr.galv2     117 non-null    int64  
 6   pr.pint      117 non-null    int64  
 7   linea        117 non-null    object 
 8   hora         117 non-null    object 
 9   temperatura  117 non-null    object 
 10  averias      117 non-null    object 
 11  naverias     117 non-null    int64  
 12  sistema      117 non-null    object 
 13  ProdTotal    117 non-null    int64  
 14  NOx          117 non-null    float64
 15  CO           117 non-null    float64
 16  COV          117 non-null    float64
 17  SO2          117 non-null    float64
 18  CO2          117 non-null    float64
 19  N2O     

In [26]:
#estadistica descriptiva con la funcion describe
df.describe()

consumo        pr.tbc        pr.cc       pr.ca    pr.galv1  \
count  117.000000    117.000000   117.000000  117.000000  117.000000   
mean   139.456496   7567.820513   295.529915  124.478632  402.803419   
std     55.185253   3002.703249   358.618556  161.990772  334.928923   
min     17.500000      0.000000     0.000000    0.000000    0.000000   
25%     99.090000   5950.000000     0.000000    0.000000    0.000000   
50%    140.070000   8808.000000    80.000000    0.000000  432.000000   
75%    182.480000   9734.000000   582.000000  248.000000  698.000000   
max    290.720000  10979.000000  1204.000000  677.000000  982.000000   

          pr.galv2     pr.pint    naverias     ProdTotal         NOx  \
count   117.000000  117.000000  117.000000    117.000000  117.000000   
mean   1159.717949  188.555556    0.675214  11841.649573    1.196132   
std     577.791602  289.445857    1.292078   4075.068276    0.643837   
min       0.000000    0.000000    0.000000   2187.000000    0.012500   
25%     933.000000    0.000000    0.000000   9023.000000    0.775000   
50%    1360.000000    0.000000    0.000000  11860.000000    1.120000   
75%    1567.000000  394.000000    0.000000  14908.000000    1.655000   
max    1963.000000  898.000000    4.000000  23202.000000    2.775000   

               CO         COV         SO2         CO2         N2O  
count  117.000000  117.000000  117.000000  117.000000  117.000000  
mean     4.377350    0.620983    0.066504  104.628141    6.114786  
std      2.095038    0.255447    0.022064   41.389709    2.125250  
min      0.250000    0.025000    0.001000   14.285000    0.870000  
25%      2.927500    0.452500    0.054000   73.717500    4.710000  
50%      4.107500    0.625000    0.066000  105.307500    6.170000  
75%      5.585000    0.785000    0.080000  134.877500    7.580000  
max     10.170000    1.255000    0.127000  218.307500   11.140000

### Interpretación de la Salida de `df.describe()`

La función `describe()` proporciona un resumen de la tendencia central, la dispersión y la forma de la distribución de las columnas numéricas en el DataFrame.

**Observaciones Generales:**

- El conjunto de datos contiene 117 entradas para todas las columnas numéricas.
- Las columnas `consumo`, `pr.tbc`, `pr.cc`, `pr.ca`, `pr.galv1`, `pr.galv2`, `pr.pint`, `naverias`, `ProdTotal`, `NOx`, `CO`, `COV`, `SO2`, `CO2` y `N2O` son numéricas y se han calculado sus estadísticas descriptivas.

**Interpretación por Columna:**

- **`consumo`**:
    - **Media (139.46):** El consumo promedio es de alrededor de 139.46.
    - **Desviación estándar (55.19):** Existe una dispersión moderada en los valores de consumo.
    - **Mínimo (17.50) y Máximo (290.72):** El consumo varía de 17.50 a 290.72.
    - **25%, 50% (Mediana), 75%:** Los cuartiles muestran la distribución de los valores de consumo. La mediana es 140.07, que está cerca de la media, lo que sugiere una distribución relativamente simétrica.

- **`pr.tbc`**:
    - **Media (7567.82):** El valor promedio para `pr.tbc` es aproximadamente 7567.82.
    - **Desviación estándar (3002.70):** Existe una dispersión significativa en los valores de `pr.tbc`.
    - **Mínimo (0.00) y Máximo (10979.00):** Los valores varían de 0 a 10979.
    - **25% (5950.00), 50% (8808.00), 75% (9734.00):** La mediana es 8808.00, mayor que la media, lo que podría indicar un ligero sesgo hacia valores más bajos.

- **`pr.cc`**:
    - **Media (295.53):** El valor promedio para `pr.cc` es de aproximadamente 295.53.
    - **Desviación estándar (358.62):** Existe una gran dispersión en los valores de `pr.cc`, en relación con la media.
    - **Mínimo (0.00) y Máximo (1204.00):** Los valores varían de 0 a 1204.
    - **25% (0.00), 50% (80.00), 75% (582.00):** El percentil 25 es 0, lo que indica que al menos una cuarta parte de las entradas tienen un valor de 0 para `pr.cc`. La mediana es 80 y la media es significativamente mayor, lo que sugiere un sesgo a la derecha.

- **`pr.ca`**:
    - **Media (124.48):** El valor promedio para `pr.ca` es de alrededor de 124.48.
    - **Desviación estándar (161.99):** Existe una dispersión considerable en los valores de `pr.ca`.
    - **Mínimo (0.00) y Máximo (677.00):** Los valores varían de 0 a 677.
    - **25% (0.00), 50% (0.00), 75% (248.00):** Los percentiles 25 y 50 son 0, lo que indica que al menos la mitad de las entradas tienen un valor de 0 para `pr.ca`. Esto sugiere un fuerte sesgo a la derecha y potencialmente muchos valores cero.

- **`pr.galv1`**:
    - **Media (402.80):** El valor promedio para `pr.galv1` es de aproximadamente 402.80.
    - **Desviación estándar (334.93):** Existe una dispersión sustancial en los valores de `pr.galv1`.
    - **Mínimo (0.00) y Máximo (982.00):** Los valores varían de 0 a 982.
    - **25% (0.00), 50% (432.00), 75% (698.00):** El percentil 25 es 0, lo que sugiere que una parte de las entradas tienen un valor de 0. La mediana (432) está cerca de la media (402.80), lo que indica una distribución relativamente simétrica entre los valores distintos de cero.

- **`pr.galv2`**:
    - **Media (1159.72):** El valor promedio para `pr.galv2` es de aproximadamente 1159.72.
    - **Desviación estándar (577.79):** Existe una dispersión moderada en los valores de `pr.galv2`.
    - **Mínimo (0.00) y Máximo (1963.00):** Los valores varían de 0 a 1963.
    - **25% (933.00), 50% (1360.00), 75% (1567.00):** La mediana (1360) es mayor que la media (1159.72), lo que podría indicar un ligero sesgo hacia valores más bajos o valores atípicos en el extremo inferior.

- **`pr.pint`**:
    - **Media (188.56):** El valor promedio para `pr.pint` es de aproximadamente 188.56.
    - **Desviación estándar (289.45):** Existe una gran dispersión en los valores de `pr.pint`, en relación con la media.
    - **Mínimo (0.00) y Máximo (898.00):** Los valores varían de 0 a 898.
    - **25% (0.00), 50% (0.00), 75% (394.00):** Los percentiles 25 y 50 son 0, lo que indica que al menos la mitad de las entradas tienen un valor de 0 para `pr.pint`. Esto sugiere un fuerte sesgo a la derecha y muchos valores cero.

- **`naverias`**:
    - **Media (0.68):** El número promedio de fallas es de aproximadamente 0.68.
    - **Desviación estándar (1.29):** Existe una dispersión moderada en el número de fallas.
    - **Mínimo (0.00) y Máximo (4.00):** El número de fallas varía de 0 a 4.
    - **25% (0.00), 50% (0.00), 75% (0.00):** Los percentiles 25, 50 y 75 son 0, lo que indica que la gran mayoría de las entradas tienen 0 fallas. Esto sugiere una distribución altamente sesgada con algunos casos de múltiples fallas.

- **`ProdTotal`**:
    - **Media (11841.65):** La producción total promedio es de alrededor de 11841.65.
    - **Desviación estándar (4075.07):** Existe una dispersión significativa en los valores de producción total.
    - **Mínimo (2187.00) y Máximo (23202.00):** La producción total varía de 2187 a 23202.
    - **25% (9023.00), 50% (11860.00), 75% (14908.00):** La mediana (11860) está muy cerca de la media (11841.65), lo que sugiere una distribución relativamente simétrica.

- **`NOx`**:
    - **Media (1.20):** La emisión promedio de NOx es de aproximadamente 1.20.
    - **Desviación estándar (0.64):** Existe una dispersión moderada en los valores de emisión de NOx.
    - **Mínimo (0.01) y Máximo (2.78):** Las emisiones de NOx varían de 0.01 a 2.78.
    - **25% (0.78), 50% (1.12), 75% (1.66):** La mediana (1.12) está cerca de la media (1.20), lo que indica una distribución relativamente simétrica.

- **`CO`**:
    - **Media (4.38):** La emisión promedio de CO es de aproximadamente 4.38.
    - **Desviación estándar (2.10):** Existe una dispersión moderada en los valores de emisión de CO.
    - **Mínimo (0.25) y Máximo (10.17):** Las emisiones de CO varían de 0.25 a 10.17.
    - **25% (2.93), 50% (4.11), 75% (5.59):** La mediana (4.11) está cerca de la media (4.38), lo que sugiere una distribución relativamente simétrica.

- **`COV`**:
    - **Media (0.62):** La emisión promedio de COV es de aproximadamente 0.62.
    - **Desviación estándar (0.26):** Existe una dispersión relativamente pequeña en los valores de emisión de COV.
    - **Mínimo (0.03) y Máximo (1.26):** Las emisiones de COV varían de 0.03 a 1.26.
    - **25% (0.45), 50% (0.63), 75% (0.79):** La mediana (0.63) está muy cerca de la media (0.62), lo que sugiere una distribución relativamente simétrica.

- **`SO2`**:
    - **Media (0.07):** La emisión promedio de SO2 es de aproximadamente 0.07.
    - **Desviación estándar (0.02):** Existe una dispersión relativamente pequeña en los valores de emisión de SO2.
    - **Mínimo (0.00) y Máximo (0.13):** Las emisiones de SO2 varían de 0.00 a 0.13.
    - **25% (0.05), 50% (0.07), 75% (0.08):** La mediana (0.07) está muy cerca de la media (0.07), lo que sugiere una distribución relativamente simétrica.

- **`CO2`**:
    - **Media (104.63):** La emisión promedio de CO2 es de aproximadamente 104.63.
    - **Desviación estándar (41.39):** Existe una dispersión significativa en los valores de emisión de CO2.
    - **Mínimo (14.29) y Máximo (218.31):** Las emisiones de CO2 varían de 14.29 a 218.31.
    - **25% (73.72), 50% (105.31), 75% (134.88):** La mediana (105.31) está muy cerca de la media (104.63), lo que sugiere una distribución relativamente simétrica.

- **`N2O`**:
    - **Media (6.11):** La emisión promedio de N2O es de aproximadamente 6.11.
    - **Desviación estándar (2.13):** Existe una dispersión moderada en los valores de emisión de N2O.
    - **Mínimo (0.87) y Máximo (11.14):** Las emisiones de N2O varían de 0.87 a 11.14.
    - **25% (4.71), 50% (6.17), 75% (7.58):** La mediana (6.17) está muy cerca de la media (6.11), lo que sugiere una distribución relativamente simétrica.

Aquí tienes algunas sugerencias para continuar con el análisis del consumo energético de la empresa productora de acero:

1.  **Limpieza y preprocesamiento de datos:**
    *   Verificar si hay valores atípicos (outliers) en las columnas numéricas, especialmente en `consumo` y las variables de producción. Considerar estrategias para manejarlos (eliminación, transformación o imputación).
    *   Confirmar que las columnas categóricas (`linea`, `hora`, `temperatura`, `averias`, `sistema`) estén correctamente codificadas o si necesitan ser convertidas a un formato numérico para ciertos análisis (por ejemplo, usando one-hot encoding).

2.  **Análisis de la relación entre el consumo energético y las variables de producción:**
    *   Realizar análisis de correlación entre `consumo` y las diferentes variables de producción (`pr.tbc`, `pr.cc`, `pr.ca`, `pr.galv1`, `pr.galv2`, `pr.pint`, `ProdTotal`). Esto ayudará a identificar qué producciones tienen una mayor relación lineal con el consumo energético.
    *   Visualizar estas relaciones utilizando gráficos de dispersión (`scatter plots`) para entender mejor la naturaleza de la relación (lineal, no lineal, etc.).

3.  **Análisis del impacto de las variables categóricas en el consumo energético:**
    *   Realizar análisis comparativos del consumo energético (`consumo`) en función de las categorías de `linea`, `hora`, `temperatura`, `averias` y `sistema`. Se pueden usar gráficos de cajas (`box plots`) o gráficos de barras (`bar plots`) para visualizar estas comparaciones.
    *   Considerar pruebas estadísticas (como ANOVA) para determinar si existen diferencias significativas en el consumo energético entre los diferentes grupos definidos por las variables categóricas.

4.  **Análisis de la influencia de las averías en el consumo y la producción:**
    *   Examinar la relación entre `averias`, `naverias` y `consumo`. ¿El número o la presencia de averías afecta el consumo energético?
    *   Analizar cómo las averías impactan en la `ProdTotal` y en las producciones individuales.

5.  **Modelado predictivo:**
    *   Si el objetivo es predecir el consumo energético, se puede construir un modelo de regresión (por ejemplo, regresión lineal, árboles de decisión, random forest) utilizando las variables de producción, las variables categóricas (previamente codificadas) y las variables relacionadas con las averías como predictores.
    *   Evaluar el rendimiento del modelo utilizando métricas apropiadas (como R-cuadrado, RMSE, MAE).

6.  **Análisis de las emisiones (NOx, CO, COV, SO2, CO2, N2O):**
    *   Aunque el enfoque principal es el consumo energético, también se pueden analizar las emisiones en relación con la producción y el consumo.
    *   Realizar análisis de correlación y visualizaciones para entender cómo varían las emisiones con la producción y otros factores.

Para implementar estas sugerencias, se pueden utilizar bibliotecas de Python como `pandas` para la manipulación de datos, `matplotlib` y `seaborn` para la visualización, y `scikit-learn` para el modelado predictivo.